packages/run-protocol/src/vaultFactory/vaultFactory.js

```diff
@@ -108,6 +108,24 @@ export const start = async (zcf, privateArgs) => {
 
   /** @type {AddVaultType} */
   const addVaultType = async (collateralIssuer, collateralKeyword, rates) => {
+    const amountPattern = { brand: M.remotable(), value: M.any() };
+    const ratioPattern = {
+      numerator: amountPattern,
+      denominator: amountPattern,
+    };
+    fit(
```

Comment:
Why does this help?  (`fit` is new for me.)  As I read the doc for `fit`, it looks like it returns a boolean.  What causes it to act as a guard here?

Reply:
The doc for `fit` might be a bit spread out. It's defined as:

```
  const fit = (specimen, patt) => {
    checkMatches(specimen, patt, assertChecker);
  };
```

... where [assertChecker](https://github.com/endojs/endo/blob/master/packages/marshal/src/helpers/passStyle-helpers.js#L50) comes from `@endo/marshal`. The doc for its type, [Checker](https://github.com/endojs/endo/blob/master/packages/marshal/src/types.js#L228-L240) should help:

```
 * @callback Checker
 * Internal to a useful pattern for writing checking logic
 * (a "checkFoo" function) that can be used to implement a predicate
 * (an "isFoo" function) or a validator (an "assertFoo" function).
 *
 *    * A predicate ideally only returns `true` or `false` and rarely throws.
 *    * A validator throws an informative diagnostic when the predicate
 *      would have returned `false`, ...
```